raiden-cli/src/index.ts

```diff
@@ -0,0 +1,311 @@
+
+    if (value && target) {
+      const availability = await raiden.getAvailability(target);
+      if (!availability.available) {
+        log.error('Target', target, 'not available:', availability);
+      } else {
+        await raidenTransfer(token, target, value);
+      }
+    }
+  }
+  const udcToken = await raiden.userDepositTokenAddress();
+  const amount = parseEther('10');
+  if ((await raiden.getTokenBalance(udcToken)).isZero()) {
+    await raiden.mint(udcToken, amount);
+    await raiden.depositToUDC(amount);
+  }
+  log.warn('Init: took', Date.now() - start);
+}
+
+async function setupApi(port: number) {
+  const api = '/api/v1';
+  app = express();
+  app.use(express.json());
+
+  app.get(`${api}/channels`, ({}, res) => {
+    raiden.channels$.pipe(first()).subscribe(c => res.json(c));
+  });
+
+  app.get(`${api}/address`, ({}, res) => {
+    res.json({ our_address: raiden.address });
+  });
+
+  app.post(`${api}/payments/:token/:target`, (req, res) => {
+    raidenTransfer(req.params.token, req.params.target, req.body.amount).then(
+      tr => res.json(tr),
+      err => res.status(400).json(err),
+    );
+  });
+
+  app.post(`${api}/config`, (req, res) => {
+    raiden.updateConfig(req.body);
+    res.json(raiden.config);
+  });
+
+  return new Promise<{ app: Express; server: Server }>(
+    resolve =>
+      (server = app.listen(port, () => {
+        log.info(`Serving Raiden LC API at http://localhost:${port}...`);
+        resolve({ app, server });
+      })),
+  );
+}
+
+async function main() {
+  logging.setLevel(logging.levels.DEBUG);
+  const argv = yargs
+    .usage('Usage: $0 -k <private_json_path> -e <node_url> --serve <port>')
+    .options({
+      privateKey: {
+        type: 'string',
+        demandOption: true,
+        alias: 'k',
+        desc: 'JSON Private Key file path',
+        coerce: path.resolve,
+      },
+      password: {
+        type: 'string',
+        desc:
+          'JSON Private Key password. Better passed through "RAIDEN_PASSWORD" env var. Prompted if not provided',
+      },
+      ethNode: {
+        alias: 'e',
+        type: 'string',
+        default: 'http://parity.goerli.ethnodes.brainbot.com:8545',
+        desc: 'ETH JSON-RPC URL',
+      },
+      store: {
+        alias: 's',
+        type: 'string',
+        default: './storage',
+        desc: 'Dir path where to store state',
+      },
+      config: {
+        alias: 'c',
+        coerce: JSON.parse,
+        desc: 'JSON to overwrite default/curretn config',
+      },
+      token: {
+        type: 'string',
+        desc: 'Token address to operate on',
+      },
+      partner: {
+        type: 'string',
+        desc: 'Open a channel with given partner',
+        implies: 'token',
+      },
+      deposit: {
+        type: 'string',
+        desc: 'Deposit this value to channel with partner',
+        implies: 'partner',
+      },
+      mint: {
```

Comment:
I would argue that in the future minting/opening channels would make more sense as something exposed through the Rest API instead of the CLI

Reply:
Yes, there's a paragraph about this on the README, for now, we expose only a limited set of features either on cli parameters and on REST API.
This one is definitely something that I want exposed in the API too, but having open + mint + deposit + 1-transfer exposed through arguments allow a specific usage of the CLI: integration testing, where we want it to sync on the chain, perform some operations adn exit.
But as soon as we have more time to dedicate to this API, of course I want all the featureset exposed through REST. If you have time to work on this, it'd be great ;)